pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -217,15 +217,20 @@ func (r *shootReconciler) runDeleteShootFlow(ctx context.Context, o *operation.O
 			Fn:           botanist.UpdateAdvertisedAddresses,
 			Dependencies: flow.NewTaskIDs(waitUntilKubeAPIServerServiceIsReady),
 		})
+		initializeSecretsManagement = g.Add(flow.Task{
+			Name:         "Initializing secrets management",
+			Fn:           flow.TaskFn(botanist.InitializeSecretsManagement).DoIf(nonTerminatingNamespace).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(ensureShootStateExists),
+		})
 		generateSecrets = g.Add(flow.Task{
 			Name:         "Generating secrets and saving them into ShootState",
 			Fn:           flow.TaskFn(botanist.GenerateAndSaveSecrets).DoIf(nonTerminatingNamespace),
-			Dependencies: flow.NewTaskIDs(ensureShootStateExists),
+			Dependencies: flow.NewTaskIDs(ensureShootStateExists, initializeSecretsManagement),
```

Comment:
`GenerateAndSaveSecrets` is still persisting those secrets not yet migrated to the new manager to the `ShootState`.

Reply:
I meant `initializeSecretsManagement` depends on `ensureShootStateExists`.
Therefore anything depending on `initializeSecretsManagement` will implicitly depend on `ensureShootStateExists`

Reply:
Yeah, it's not strictly needed, however, without this dependency the CA secrets might be created with the `persist=true` label before the `ShootState` exists. This can yield to confusing error logs when the added gardenlet reconciler tries to persist them.

Reply:
But I see we do this all the time in the Flow that we depend on something even though it is implicitly already depended on.

Reply:
Ah, I didn't get your point - it was about removing the explicitness. But yes, let's keep it like this to make it more clear.